### PR TITLE
feat(pack): added docker pack

### PR DESCRIPTION
**Pull request by Souravpakhira**
Created docker folder inside pack. Added dockerfile-language-server and docker-compose-language-service language servers for LSP and hadolint for linting support of dockerfile and docker-compose file.

### CODE REVIEW

File: lua/astrocommunity/pack/docker/docker.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
+local utils = require "astronvim.utils"
+return {
+  {
+    "williamboman/mason-lspconfig.nvim",
+    opts = function(_, opts)
+      opts.ensure_installed =
+        utils.list_insert_unique(opts.ensure_installed, "docker-compose-language-service", "dockerfile-language-server")
```
Comment (mehalter) [inline lua/astrocommunity/pack/docker/docker.lua:7]:
```suggestion
        utils.list_insert_unique(opts.ensure_installed, { "docker-compose-language-service", "dockerfile-language-server" })
```

this should be a list like table

**Review by mehalter — CHANGES_REQUESTED** [review]
Made a suggestion to fix a bug. You should also add `dockerfile` to the ensure installation for `nvim-treesitter`

File: lua/astrocommunity/pack/docker/docker.lua (hunk 1 above)
Comment (Souravpakhira) [reply lua/astrocommunity/pack/docker/docker.lua]: added as suggested

File: lua/astrocommunity/pack/docker/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Docker Language Pack
+
+This plugin pack does the following:
+
+- Adds `dockerfile-language-server` and `docker-compose-language-service` language servers
```
Comment (mehalter) [inline lua/astrocommunity/pack/docker/README.md:5]: could you add the treesitter modification as well to the README

**Review by mehalter — CHANGES_REQUESTED** [review]
After the README update this looks perfect! Thanks so much for the contribution!

**Pull request merged**